src/Model/Behavior/RegisterBehavior.php

```diff
@@ -63,6 +63,7 @@ public function register($user, $data, $options)
         //@todo move updateActive to afterSave?
         $user = $this->_updateActive($user, $validateEmail, $tokenExpiration);
         $this->_table->isValidateEmail = $validateEmail;
+        $this->_table->usernameBlackList = Configure::read('Users.Username.blackList', []);
```

Comment:
you can read the configuration from inside the UsersTable:203 callback and leave this trait unchanged